src/http-client/node-http2-client.ts

```diff
@@ -213,4 +221,107 @@ export class NodeHTTP2Client implements HTTPClient {
       }
     });
   }
+
+  /** {@inheritDoc HTTPStreamClient.stream} */
+  #doStream({
+    data: requestData,
+    headers: requestHeaders,
+    method,
+  }: HTTPStreamRequest): StreamAdapter {
+    let resolveChunk: (chunk: string) => void;
+    let rejectChunk: (reason: any) => void;
+
+    const setChunkPromise = () =>
+      new Promise<string>((res, rej) => {
+        resolveChunk = res;
+        rejectChunk = rej;
+      });
+
+    let chunkPromise = setChunkPromise();
+
+    let req: ClientHttp2Stream;
+    const onResponse = (
+      http2ResponseHeaders: IncomingHttpHeaders & IncomingHttpStatusHeader
+    ) => {
+      const status = Number(
+        http2ResponseHeaders[http2.constants.HTTP2_HEADER_STATUS]
+      );
+      if (!(status >= 200 && status < 400)) {
+        // TODO: if we get bad status, then we should still finish reading the response, and create
+        // the appropriate error instance
```

Comment:
We need to branch here to finish reading the data and throw an appropriate Fauna error, instead of trying to yield any events to the stream.

Reply:
The branching logic is done. As a follow-up, we should let the Client do the work to identify what kind of error was received.